### DISCUSSION TITLE
Tutorial - Range testing and mapping

### DISCUSSION THREAD

**Discussion by timmbogner** [timmbogner/Farm-Data-Relay-System#161] [Show and tell]
I made[ this tutorial ](https://hackaday.io/project/190006-range-testing-with-fdrs-and-node-red)that uses the GPS example in conjunction with the Node-RED Worldmap node.

This would be good spot to post any issues or suggestions.

**Comment by bvwelch** [comment]
Great tutorial!  It answers a lot of questions that I had.

Question: what LORA parameters did you use?  bandwidth, spreading factor, etc?

Based on some quick searching, I think your tutorial uses: bw=125, spread=5, cr=5, pwr=17.

I am trying to learn to use this calculator:    https://lora-developers.semtech.com/build/tools/calculator

based on reading this article:   https://stuartsprojects.github.io/2020/05/07/How-fast-is-LoRa.html

**Reply by timmbogner** [reply]
Thanks! I didn't add extra configuration because I worried complicating the instructions, but I may add an extra step explaining how to switch LoRa configurations and also how to use ESP-NOW in long-range mode.
The default LoRa parameters are the ones found in [fdrs_globals.h](https://github.com/timmbogner/Farm-Data-Relay-System/blob/main/src/fdrs_globals.h). You can edit the copy in your Arduino libraries folder to change the settings, or set them within your sketch by removing "GLOBAL_" from the macro name. For example: ```#define LORA_FREQUENCY 868.0``` would override the global setting.
The defaults are:
``` cpp
#define GLOBAL_LORA_FREQUENCY 915.0   // Carrier frequency in MHz. Allowed values range from 137.0 MHz to 1020.0 MHz (varies by chip).
#define GLOBAL_LORA_SF 7     // LoRa link spreading factor. Allowed values range from 6 to 12.
#define GLOBAL_LORA_BANDWIDTH 125.0  // LoRa link bandwidth in kHz. Allowed values are 10.4, 15.6, 20.8, 31.25, 41.7, 62.5, 125, 250 and 500 kHz.
#define GLOBAL_LORA_CR 5    // LoRa link coding rate denominator. Allowed values range from 5 to 8.
#define GLOBAL_LORA_SYNCWORD 0x12    // LoRa sync word. Can be used to distinguish different LoRa networks. Note that 0x34 is reserved for LoRaWAN.
#define GLOBAL_LORA_INTERVAL 5000  // Interval between LoRa buffer releases. Must be longer than transmission time-on-air.
```
There is also transmission power, which is not globally defined, but found in config files:
``` cpp
#define LORA_TXPWR 17   // LoRa TX power in dBm (: +2dBm - +17dBm (for SX1276-7) +20dBm (for SX1278))
```
